### PR TITLE
Bump MSRV to 1.61

### PR DESCRIPTION
**Pull request by djc**
This is used in #1508 and #1414, so probably makes sense to pull this out separately and get it merged sooner.

Addressed comments from @cpu in #1508.

### CODE REVIEW

File: connect-tests/Cargo.toml [hunk 1]
```diff
@@ -2,7 +2,6 @@
 name = "rustls-connect-tests"
 version = "0.0.1"
 edition = "2021"
-rust-version = "1.60"
```
Comment (cpu) [inline connect-tests/Cargo.toml:5]: Can you expand on the rationale for this for my own edification? I don't think it's the wrong change, I just want to understand it better.

My previous comment said:
> The [workspace docs](https://doc.rust-lang.org/cargo/reference/workspaces.html#the-package-table) read to me like we need a [workspace.package] in the top-level Cargo.toml for this to be inherited, is that incorrect?

Was that incorrect? Does the `rust-version` get inherited even in the absence of the top level `workspace.package`? Is this somehow related to the MSRV change or just tidying?

**Review by cpu — APPROVED** [review]

File: connect-tests/Cargo.toml (hunk 1 above)
Comment (djc) [reply connect-tests/Cargo.toml:5]: No, there's no implicit inheritance: only if we used `rust-version = { workspace: true }` or `rust-version.workspace = true` would we inherit it. It's really only cleaning up version tracking that we don't actually use before we bump it in the places we do use it.

Comment (cpu) [reply connect-tests/Cargo.toml:5]: :+1: That makes sense, thank you! I think the main intuition I was missing here is that there's no value in configuring an MSRV for crates internal to the project that aren't published externally.

**Review by ctz — APPROVED** [review]
